### PR TITLE
Add climb code

### PR DESCRIPTION
**Pull request by yoshiyahoo**
ive added some basic code for the climbing system

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
 
 public class Climb extends SubsystemBase {
 
+    public WPI_TalonSRX elevatorMotor = new WPI_TalonSRX(Constants.ClimbConstants.CLIMB_ELEVATOR_ID);
+    public WPI_TalonSRX winchMotor = new WPI_TalonSRX(Constants.ClimbConstants.CLIMB_WINCH_ID);
+
     public Climb() {
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java:14]: add set to factory default for both your motors

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 2]
```diff
@@ -1,30 +1,44 @@
 package frc.robot.subsystems;
 
+import com.ctre.phoenix.motorcontrol.LimitSwitchNormal;
+import com.ctre.phoenix.motorcontrol.RemoteLimitSwitchSource;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: add '.Climb' to the end of this import

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 3]
```diff
@@ -1,30 +1,44 @@
...
     }
 
     public void spinWinchForward() {
-
+        winchMotor.set(0.3);
     }
 
     public void spinWinchBackward() {
-
+        winchMotor.set(0.3);
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: these are both +

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 4]
```diff
@@ -1,30 +1,44 @@
...
     public Climb() {
+        elevatorMotor.configForwardLimitSwitchSource(RemoteLimitSwitchSource.RemoteTalonSRX, LimitSwitchNormal.NormallyOpen,Constants.ClimbConstants.CLIMB_ELEVATOR_ID,0);
+        elevatorMotor.configReverseLimitSwitchSource(RemoteLimitSwitchSource.RemoteTalonSRX, LimitSwitchNormal.NormallyClosed,Constants.ClimbConstants.CLIMB_ELEVATOR_ID,0);
 
+        winchMotor.configForwardLimitSwitchSource(RemoteLimitSwitchSource.RemoteTalonSRX, LimitSwitchNormal.NormallyOpen,Constants.ClimbConstants.CLIMB_WINCH_ID,0);
     }
 
     public void goUp() {
-
+        if(elevatorMotor.isFwdLimitSwitchClosed() != 1){
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: add tge 1 to the constants file

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
+            elevatorMotor.set(0.3);
+        }
     }
 
     public void goDown() {
-
+        if(elevatorMotor.isRevLimitSwitchClosed() != 1){
+            elevatorMotor.set(-0.3);
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: same with these power outputs

**Review by SOR-FRCTeam6059 — DISMISSED** [review]

**Review by Palak-Guleria — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 6]
```diff
@@ -35,7 +38,7 @@ public void spinWinchForward() {
     }
 
     public void spinWinchBackward() {
-        winchMotor.set(0.3);
+        winchMotor.set(-0.3);
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: add these to the constants

**Review by SOR-FRCTeam6059 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 7]
```diff
@@ -29,16 +29,16 @@ public void goUp() {
...
     }
 
     public void spinWinchForward() {
-        winchMotor.set(0.3);
+        winchMotor.set(ClimbConstants.WINCH_POWER_CONSTANT);
     }
 
     public void spinWinchBackward() {
-        winchMotor.set(-0.3);
+        winchMotor.set(-ClimbConstants.WINCH_POWER_CONSTANT);
```
Comment (SOR-FRCTeam6059) [inline src/main/java/frc/robot/subsystems/Climb.java]: have this as a seperate constant incase the forward speed needs to be less or more than the backwards

**Review by SOR-FRCTeam6059 — DISMISSED** [review]

**Review by Palak-Guleria — APPROVED** [review]

**Review by Palak-Guleria — APPROVED** [review]

**Pull request merged**